packages/block-editor/src/components/use-block-sync/block-manager.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { v4 as uuid } from 'uuid';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+function createBlockClientIdManager( shouldMapBlocks ) {
+	const DIRECTION_IN = 'in';
+	const DIRECTION_OUT = 'out';
+	const flipDirection = ( direction ) => {
+		return direction === DIRECTION_IN ? DIRECTION_OUT : DIRECTION_IN;
+	};
```

Comment:
We could optimize this a bit by defining these objects outside of `createBlockClientIdManager`, so they aren't duplicated every time the function is run. They're not specific to any instance, so it should be safe to do so.